src/OpenCL.cpp

```diff
@@ -49,7 +49,7 @@ static std::string sourceCode_config = R"(
     typedef float net_t;
     #define vload_net_t(offset,p) ((p)[(offset)])
     #define vstore_net_t(data,offset,p) (((p)[(offset)])=(data))
-)";
+    #define BOARD_SIZE )" + std::to_string(BOARD_SIZE);
```

Comment:
This doesn't look right

Reply:
It's the neatest way I've found in order to pass the BOARD_SIZE define into the OpenCl code source (the code doesn't have access to external defines). My aim with this is so the user only has to modify the board size in one place (namely config.h).

I'm ready to change this to some other thing if you have a better idea, but I don't have any other idea myself if I want to keep this "only has to modify the size in config.h" constraint.